modules/repository-datastax/src/test/java/com/intuit/wasabi/repository/cassandra/impl/AssignmentCountEnvelopeTest.java

```diff
@@ -77,7 +77,7 @@ public void testRapidExperimentationLimitNotReached() {
 //        doReturn(42).when(exp).getUserCap();
         when(exp.getIsRapidExperiment()).thenReturn(true);
         when(exp.getUserCap()).thenReturn(42);
-        when(ar.getBucketAssignmentCount(exp).getTotalUsers().getBucketAssignments()).thenReturn(41l);
+        when(ar.getBucketAssignmentCount(exp).getTotalUsers().getBucketAssignments()).thenReturn(40l);
```

Comment:
@iizrailevsky Earlier the check for cap used to happen after updating the bucket count. With the latest code changes, it will happen before the update. So, there is a difference of 1 in the count which needs to be modified in the test. We are verifying "limit not reached here". If the limit was "42" and this method returns "41" already, and the check happens for "userCap - 1" before updating the count, it will say "limit reached". Hope it makes sense.